### PR TITLE
Fix random test failure in test_mul_within_range

### PR DESCRIPTION
**Pull request by seanyoung**
If gen_biguint() generated a 0, then the sub(1) will cause the test to fail.

See https://github.com/hyperledger/solang/actions/runs/3044527275/jobs/4905014572

### CODE REVIEW

File: tests/substrate_tests/expressions.rs [hunk 1]
```diff
@@ -1119,8 +1119,8 @@ fn test_mul_within_range() {
         let width_rounded = (width as usize / 8).next_power_of_two();
         let mut runtime = build_solidity(&src);
 
-        // The range of values that can be held in signed N bits is [-2^(N-1), 2^(N-1)-1]. Here we generate a random number within this range and multiply it by 1
-        let a = rng.gen_biguint((width).try_into().unwrap()).sub(1_u32);
+        // The range of values that can be held in unsigned N bits is [0, 2^N-1]. Here we generate a random number within this range and multiply it by 1
+        let a = rng.gen_biguint((width).try_into().unwrap());
```
Comment (LucasSte) [inline tests/substrate_tests/expressions.rs:1123]: I agree that this test needs fixing, but how are we testing negative values now?

**Comment by seanyoung** [comment]
I've been running:

```bash
$ while cargo test --test substrate --  --exact  substrate_tests::expressions::test_mul_within_range ; do : ; done
```

For a few hours now, no failures any more

File: tests/substrate_tests/expressions.rs (hunk 1 above)
Comment (seanyoung) [reply tests/substrate_tests/expressions.rs:1123]: This is the unsigned test

**Review by LucasSte — APPROVED** [review]

**Review by xermicus — APPROVED** [review]
